src/taco-dependency-installer/test/androidSdkInstaller.ts

```diff
@@ -0,0 +1,131 @@
+            // Register mocks. child_process and taco-utils mocks needs to be register before 
+            // AndroidSdkInstaller is required for the mock to work
+            mockery.registerMock("child_process", new nodeFakes.ChildProcessModule().fakeAllExecCallsEndingWithErrors());
+
+            var fakeTelemetryHelper: TacoTestsUtils.TelemetryFakes.Helper = new tacoTestsUtils.TelemetryFakes.Helper();
+            var tacoUtilsWithFakeTelemetry = _.extend({}, tacoUtils, { TelemetryHelper: fakeTelemetryHelper, HasFakes: true });
+            mockery.registerMock("taco-utils", tacoUtilsWithFakeTelemetry);
+
+            // We require the AndroidSdkInstaller file, which will use all the mocked dependencies
+            var androidSdkInstallerClass = require("../installers/androidSdkInstaller");
+
+            // Mocking process messes with node.js, so we do it after requiring all the files
+            var fakeProcess = new nodeFakes.Process().fakeMacOS().fakeDeterministicHrtime().asProcess();
+            // TODO: Try to find a nicer way of doing this. Maybe setting process as protoype of FakeProcess?
+            process = <any> _.extend({}, process, fakeProcess);
+
+            var steps: DependencyInstallerInterfaces.IStepsDeclaration = {
+                download: false,
+                install: false,
+                updateVariables: true, // We only test this step on this test
+                postInstall: false
+            };
+
+            var installerInfo: DependencyInstallerInterfaces.IInstallerData = {
+                installSource: "",
+                sha1: "",
+                bytes: 0,
+                installDestination: "",
+                steps: steps
+            };
+
+            var softwareVersion: string = "";
+            var installTo: string = "";
+
+            var androidSdkInstaller = new androidSdkInstallerClass(installerInfo, softwareVersion, installTo,
+                new FakeLogger(), steps);
+
+            var expectedTelemetry = [
+                {
+                    "initialStep.time": { isPii: false, value: "2000" },
+                    step: { isPii: false, value: "initialStep" }
+                },
+                {
+                    "error.description": { isPii: false, value: "ErrorOnChildProcess on updateVariablesDarwin" },
+                    lastStepExecuted: { isPii: false, value: "updateVariables" },
+                    step: { isPii: false, value: "updateVariables" },
+                    time: { isPii: false, value: "3000" },
+                    "updateVariables.time": { isPii: false, value: "2000" }
+                }
+            ];
+
+            androidSdkInstaller.run()
+                .done(() => done(new Error("Should have gotten a rejection in this test")), () => {
+                    fakeTelemetryHelper.getAllSentEvents().done((allSentEvents: TelemetryEvent[]) => {
+                        allSentEvents.should.eql(expectedTelemetry);
```

Comment:
You were right, the test wasn't working.

Updating with this code which seems to work:
            return androidSdkInstaller.run()
                .then(() => Q.reject(new Error("Should have gotten a rejection in this test")), () => {
                    return fakeTelemetryHelper.getAllSentEvents().then((allSentEvents: TelemetryEvent[]) => {
                        allSentEvents.should.eql(expectedTelemetry);
                    });
                }).done(done, done);


Now I get:
 AndroidSdkInstaller telemetry
    updateVariablesDarwin
      1) generates telemetry if there is an error on the update command


  0 passing (63ms)
  1 failing